did_core/did_parser_nom/tests/did/negative.rs

```diff
@@ -0,0 +1,27 @@
+use did_parser_nom::Did;
+
+macro_rules! test_cases_negative {
+    ($($name:ident: $input:expr)*) => {
+        $(
+            #[test]
+            fn $name() {
+                println!("Testing {}", $input);
+                assert!(Did::parse($input.to_string()).is_err());
+            }
+        )*
+    };
+}
+
+test_cases_negative! {
+    test_failure_case1: ""
+    test_failure_case2: "not-a-did"
+    test_failure_case3: "did:example"
+    test_failure_case4: "2ZHFFhtTD6hJqzux"
+    test_failure_case5: "did:indy:s@vrin:7Tqg6BwSSWapxgUDm9KKgg"
+    test_failure_case6: "did:indy:sovrin:alpha:%0zqg6BwS.Wapxg-Dm9K_gg"
+    test_failure_case7: "did:sov:2wJPyULfLLnYTEFYzByf"
+    test_failure_case8: "did:sov:2wJPyULfOLnYTEFYzByfUR"
```

Comment:
It's non-obvious why these are not valid dids - I presume some non-base58 characters? I'd suggest to make test cases more descriptive, although they won't line up here visually as pretty

Reply:
I assumed comparing against the valid cases makes it clear.

Reply:
Are you suggesting I should look up into different file to understand what is `test_failure_case7` doing?

Reply:
To understand what makes a DID invalid you should understand what makes it valid, which at the very least means looking at the positive test cases, preferably the method's specification; there is no way to explain the same in a test name.

At best, the test name can give an illusion of understanding or partial explanation (like `test_sov_unqualified_invalid_len` or `test_sov_unqualified_invalid_char`), at the cost of long, hard-to-read and potentially confusing test names (like `test_indy_non_method_specific_id_char_in_namespace`).

I understand and am sympathetic to the argument that (or test comments) is better than no explanation / documentation at all.